lib/utils/proxy/proxy.go

```diff
@@ -149,7 +158,7 @@ func (d proxyDial) DialTimeout(network, address string, timeout time.Duration) (
 		defer cancel()
 		ctx = timeoutCtx
 	}
-	conn, err := dialProxy(ctx, d.proxyHost, address)
+	conn, err := apiclient.DialProxy(ctx, d.proxyHost, address, nil)
```

Comment:
Why are we passing `nil` here?

Reply:
This tells `DialProxy` to use the default `net.Dialer` instead of a passed in dialer. Would it be clearer to always pass in the dialer, or to do something else?